examples/workflows/clustering_example.py

```diff
@@ -5,7 +5,7 @@
 
 from workflows_core.api.client import Client
 from workflows_core.dataset.dataset import Dataset
-from workflows_core.engine.stable_engine import StableEngine
```

Comment:
Why is it ClusterEngine?

Reply:
cos it specific to clustering only

Reply:
@jtwinrelevanceai  No it's not. Here are a few other use cases that require this: 
- Dimensionality Reduction 
- in-Memory clustering algorithmms (HDBSCAN, Image Clustering) 
- Any other in-memory algorithm.